### PR TITLE
feat(sql): parallelize JIT-compiled filters via async offload

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Depends on #1732

Migrates JIT-compiled filters to the new async offload API. Removes single-threaded JIT-compiled filter record factory and cursor classes.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 993 / 1078 (92.12%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|13|00.00%|
|:large_blue_circle:|io/questdb/cairo/sql/RecordCursorFactory.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/mp/FanOut.java|8|12|66.67%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByRecordCursorFactory.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/DataFrameRecordCursorFactory.java|6|8|75.00%|
|:large_blue_circle:|io/questdb/cairo/sql/PageAddressCacheRecord.java|143|177|80.79%|
|:large_blue_circle:|io/questdb/cairo/sql/NetworkSqlExecutionCircuitBreaker.java|28|32|87.50%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|24|27|88.89%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|17|19|89.47%|
|:large_blue_circle:|io/questdb/cairo/sql/PageAddressCache.java|46|50|92.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameReduceJob.java|59|63|93.65%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameSequence.java|164|174|94.25%|
|:large_blue_circle:|io/questdb/griffin/engine/table/AsyncJitFilteredRecordCursorFactory.java|108|110|98.18%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgLongVectorAggregateFunction.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/DistinctTimeSeriesRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/SqlExecutionCircuitBreaker.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/StatefulAtom.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByAllIndexedRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/MessageBusImpl.java|34|34|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/orderby/SortedRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByFirstLastRecordCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgDoubleVectorAggregateFunction.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgIntVectorAggregateFunction.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|15|15|100.00%|
|:large_blue_circle:|io/questdb/std/Misc.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3Utils.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/DeferredSingleSymbolFilterDataFrameRecordCursorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameReduceTask.java|26|26|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/AsyncFilteredRecordCursor.java|90|90|100.00%|
|:large_blue_circle:|io/questdb/std/DirectLongList.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/MessageBus.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByFillNoneRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/GroupByNotKeyedVectorRecordCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/GroupByRecordCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/jit/CompiledFilterIRSerializer.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/map/FastMap.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByInterpolateRecordCursorFactory.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/union/IntersectRecordCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/AbstractNoRecordSampleByCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/union/UnionRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameDispatchJob.java|62|62|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/udp/AbstractLineProtoUdpReceiver.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/mp/AbstractSSequence.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByFillValueRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/rnd/SharedRandom.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/DistinctRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameDispatchTask.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashOuterJoinLightRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/AbstractSampleByFillRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/GroupByNotKeyedRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashOuterJoinRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/orderby/LimitedSizeSortedLightRecordCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashJoinLightRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/AsyncFilteredRecordCursorFactory.java|32|32|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/GroupByRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/orderby/SortedLightRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/union/ExceptRecordCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByLightRecordCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByFillPrevRecordCursor.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 997 / 1081 (92.23%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|13|00.00%|
|:large_blue_circle:|io/questdb/cairo/sql/RecordCursorFactory.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/mp/FanOut.java|8|12|66.67%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByRecordCursorFactory.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/DataFrameRecordCursorFactory.java|6|8|75.00%|
|:large_blue_circle:|io/questdb/cairo/sql/PageAddressCacheRecord.java|143|177|80.79%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|24|27|88.89%|
|:large_blue_circle:|io/questdb/cairo/sql/NetworkSqlExecutionCircuitBreaker.java|27|30|90.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameReduceJob.java|54|60|90.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|19|21|90.48%|
|:large_blue_circle:|io/questdb/cairo/sql/PageAddressCache.java|46|50|92.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameSequence.java|167|174|95.98%|
|:large_blue_circle:|io/questdb/griffin/engine/table/AsyncJitFilteredRecordCursorFactory.java|108|110|98.18%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameDispatchJob.java|61|62|98.39%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgLongVectorAggregateFunction.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/DistinctTimeSeriesRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/SqlExecutionCircuitBreaker.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/StatefulAtom.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByAllIndexedRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/MessageBusImpl.java|34|34|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/orderby/SortedRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByFirstLastRecordCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgDoubleVectorAggregateFunction.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgIntVectorAggregateFunction.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/std/Misc.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3Utils.java|12|12|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/DeferredSingleSymbolFilterDataFrameRecordCursorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameReduceTask.java|26|26|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/AsyncFilteredRecordCursor.java|90|90|100.00%|
|:large_blue_circle:|io/questdb/std/DirectLongList.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/MessageBus.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByFillNoneRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/GroupByNotKeyedVectorRecordCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/GroupByRecordCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/jit/CompiledFilterIRSerializer.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/map/FastMap.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByInterpolateRecordCursorFactory.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/union/IntersectRecordCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/AbstractNoRecordSampleByCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/union/UnionRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/udp/AbstractLineProtoUdpReceiver.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/mp/AbstractSSequence.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByFillValueRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/rnd/SharedRandom.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/DistinctRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameDispatchTask.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashOuterJoinLightRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/AbstractSampleByFillRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/GroupByNotKeyedRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashOuterJoinRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/orderby/LimitedSizeSortedLightRecordCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashJoinLightRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/AsyncFilteredRecordCursorFactory.java|32|32|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/GroupByRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/orderby/SortedLightRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/union/ExceptRecordCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByLightRecordCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByFillPrevRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextQueryProcessor.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1019 / 1104 (92.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/ServerMain.java|0|13|00.00%|
|:large_blue_circle:|io/questdb/cairo/sql/RecordCursorFactory.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/mp/FanOut.java|8|12|66.67%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByRecordCursorFactory.java|3|4|75.00%|
|:large_blue_circle:|io/questdb/cairo/sql/PageAddressCacheRecord.java|143|177|80.79%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|23|26|88.46%|
|:large_blue_circle:|io/questdb/griffin/engine/table/DataFrameRecordCursorFactory.java|17|19|89.47%|
|:large_blue_circle:|io/questdb/cairo/sql/NetworkSqlExecutionCircuitBreaker.java|27|30|90.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|20|22|90.91%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameReduceJob.java|55|60|91.67%|
|:large_blue_circle:|io/questdb/cairo/sql/PageAddressCache.java|46|50|92.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameSequence.java|167|174|95.98%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameDispatchJob.java|59|62|95.16%|
|:large_blue_circle:|io/questdb/griffin/engine/table/AsyncJitFilteredRecordCursorFactory.java|108|110|98.18%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgLongVectorAggregateFunction.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/DistinctTimeSeriesRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContextImpl.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/SqlExecutionCircuitBreaker.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/StatefulAtom.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByAllIndexedRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/MessageBusImpl.java|34|34|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/orderby/SortedRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByFirstLastRecordCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgDoubleVectorAggregateFunction.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgIntVectorAggregateFunction.java|11|11|100.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|18|18|100.00%|
|:large_blue_circle:|io/questdb/std/Misc.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3Utils.java|12|12|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/DeferredSingleSymbolFilterDataFrameRecordCursorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameReduceTask.java|26|26|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/AsyncFilteredRecordCursor.java|90|90|100.00%|
|:large_blue_circle:|io/questdb/std/DirectLongList.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/MessageBus.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByFillNoneRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/std/MemoryTag.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/GroupByNotKeyedVectorRecordCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/GroupByRecordCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/jit/CompiledFilterIRSerializer.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/map/FastMap.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByInterpolateRecordCursorFactory.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/union/IntersectRecordCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlExecutionContext.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/AbstractNoRecordSampleByCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/union/UnionRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/udp/AbstractLineProtoUdpReceiver.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/mp/AbstractSSequence.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByFillValueRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/rnd/SharedRandom.java|7|7|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/JsonQueryProcessor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/DistinctRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/sql/async/PageFrameDispatchTask.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashOuterJoinLightRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/AbstractSampleByFillRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/GroupByNotKeyedRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashOuterJoinRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/orderby/LimitedSizeSortedLightRecordCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/join/HashJoinLightRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/AsyncFilteredRecordCursorFactory.java|32|32|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/GroupByRecordCursorFactory.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/orderby/SortedLightRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/union/ExceptRecordCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/LatestByLightRecordCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/SampleByFillPrevRecordCursor.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/http/processors/TextQueryProcessor.java|1|1|100.00%|

File: core/src/main/java/io/questdb/log/Logger.java [hunk 1]
```diff
@@ -358,7 +358,7 @@ private LogRecord next(Sequence seq, RingQueue<LogRecordSink> ring, int level) {
             return NullLogRecord.INSTANCE;
         }
 
-        final long cursor = seq.next();
+        final long cursor = seq.nextBully();
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/log/Logger.java:361]: @bluestreak01 I don't think this change comes from my commit. Looks like a debug leftover. WDYT?

Comment (bluestreak01) [reply core/src/main/java/io/questdb/log/Logger.java:361]: yeah, its mine, sync logger, will remove

**Pull request merged**